packages/lesswrong/server/migrations/meta/utils.ts

```diff
@@ -117,6 +117,21 @@ export const dropIndex = async <N extends CollectionNameString>(
   await db.none(sql, args);
 }
 
+export const dropIndexByName = async <N extends CollectionNameString>(
+  db: SqlClientOrTx,
+  collection: CollectionBase<N>,
+  indexName: string
+): Promise<void> => {
+  const indexes = collection.getTable().getIndexes();
+  const indexesToDrop = indexes.filter(index => index.getName() === indexName);
```

Comment:
Have you tested this? (I haven't so I could be wrong)

IIRC, `index.getName()` returns the full index name which is made by concatenating the names of the columns in the index, but when postgres creates the index there's a 63 character limit, after which any more characters get truncated.

I think that means we need to do `index.getName.slice(0, 63) === indexName`.

(Also, there's a deeper bug here where this could lead to naming collisions, but I don't remember ever actually running into that)

Reply:
I wasn't tracking this but it worked out because I got the index name by copy-pasting it out of pgadmin, where it was truncated to 63 characters, rather than by gluing together field names. However when testing it turned out that there was a different problem, in that `getIndexes()` was returning the list of requested indexes, rather than actual indexes, so (once the `ensureIndex` was removed) it didn't include the index to drop.

Reply:
Fixed by no longer relying on `getIndexes()` (and renaming it to make it clear that that function reflects what's in the code, not necessarily what's in the database).